integration/spark-common/src/main/scala/org/apache/spark/sql/execution/command/carbonTableSchemaCommon.scala

```diff
@@ -848,6 +848,19 @@ class TableNewProcessor(cm: TableModel) {
       tableSchema.getTableId,
       cm.databaseNameOp.getOrElse("default"))
     tablePropertiesMap.put("bad_record_path", badRecordsPath)
+    if (tablePropertiesMap.get("sort_columns") != null) {
+      val sortCol = tablePropertiesMap.get("sort_columns")
+      if ((!sortCol.trim.isEmpty) && tablePropertiesMap.get("sort_scope") == null) {
+        // If sort_scope is not specified, but sort_columns are present, set sort_scope as
+        // local_sort in carbon_properties (cannot add in table properties as if user sets carbon
+        // properties it won't be reflected as table properties is given higher priority)
+        if (CarbonProperties.getInstance().getProperty(CarbonCommonConstants.LOAD_SORT_SCOPE) ==
+            null) {
+          CarbonProperties.getInstance()
+            .addProperty(CarbonCommonConstants.LOAD_SORT_SCOPE, "LOCAL_SORT")
```

Comment:
I don't think we can add sort_scope directly to CarbonProperties, it changes for the complete system.
If sortColumns are present then default table sort_scope should become local_sort not to complete system

Reply:
there are test cases, that just set sort scope in carbon property but not sort columns, that test case fails as table property that I set has higher priority than carbon properties



example, sort_columns are mentioned in table properties, but sort_scope is present as "Batch_sort" in carbon properties, so in this time If I set in table properties, test case fails.

that too they set carbon properties after creating the table !

Reply:
ok. changed back to table properties